src/test/java/com/artipie/rpm/http/RpmSliceDownloadITCase.java

```diff
@@ -0,0 +1,154 @@
+import com.artipie.rpm.RepoConfig;
+import com.artipie.rpm.TestRpm;
+import com.artipie.vertx.VertxSliceServer;
+import io.vertx.reactivex.core.Vertx;
+import java.io.IOException;
+import java.util.Optional;
+import org.cactoos.list.ListOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.text.StringContainsInOrder;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.Testcontainers;
+import org.testcontainers.containers.GenericContainer;
+
+/**
+ * Integration test for {@link RpmSlice}.
+ * @since 0.10
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+final class RpmSliceDownloadITCase {
+
+    /**
+     * Vertx instance.
+     */
+    private static final Vertx VERTX = Vertx.vertx();
+
+    /**
+     * Vertx slice server instance.
+     */
+    private VertxSliceServer server;
+
+    /**
+     * Container.
+     */
+    private GenericContainer<?> cntn;
+
+    /**
+     * Port.
+     */
+    private int port;
+
+    @Test
+    void installs() throws Exception {
+        final TestRpm rpm = new TestRpm.Time();
+        this.start(rpm, Permissions.FREE, Identities.ANONYMOUS);
+        MatcherAssert.assertThat(
+            this.executeCommand(
+                String.format(
+                    "http://host.testcontainers.internal:%d/%s.rpm",
+                    this.port, rpm.name()
+                )
+            ),
+            new StringContainsInOrder(new ListOf<>(rpm.name(), "Complete!"))
+        );
+    }
+
+    @Test
+    void installsWithAuth() throws Exception {
+        final String john = "john";
+        final String pswd = "123";
+        final TestRpm rpm = new TestRpm.Time();
+        this.start(
+            rpm,
+            (name, perm) -> john.equals(name) && "download".equals(perm),
+            new BasicIdentities(
+                (name, pass) -> {
+                    final Optional<String> res;
+                    if (john.equals(name) && pswd.equals(pass)) {
+                        res = Optional.of(name);
+                    } else {
+                        res = Optional.empty();
+                    }
+                    return res;
+                }
+            )
+        );
+        MatcherAssert.assertThat(
+            this.executeCommand(
+                String.format(
+                    "http://%s:%s@host.testcontainers.internal:%d/%s.rpm", john,
+                    pswd, this.port, rpm.name()
+                )
+            ),
+            new StringContainsInOrder(new ListOf<>(rpm.name(), "Complete!"))
+        );
+    }
+
+    @AfterEach
+    void stopContainer() {
+        this.server.close();
+        this.cntn.stop();
+    }
+
+    @AfterAll
+    static void close() {
+        RpmSliceDownloadITCase.VERTX.close();
+    }
+
+    private String executeCommand(final String url) throws IOException, InterruptedException {
```

Comment:
@olenagerasimova this method name is not very clear: it claims to execute a command but actually it's running only `yum`. Maybe we can rename it to `yumInstall` to be more specific?